### DISCUSSION TITLE
InstallerSwitches for package uninstall

### DISCUSSION THREAD

**Discussion by mariXanna** [microsoft/winget-cli#1221] [Q&A] [answered]
Is it possible to pass command line options to the installer of executable type when winget is uninstalling the package?
For example, the application requires specific parameters passed to be installed and removed silently.
"InstallerSwitches" work perfectly in case for installation, but they aren't passed for uninstallation and thus uninstallation cannot be done silently. I have not find any information in winget documentation on this, using "Custom" installer switches also didn't help.

**Comment by denelon** [comment, marked as answer]
@mariXanna no, we don't have an "--override" argument equivalent for `winget uninstall`. https://github.com/microsoft/winget-cli/issues/1089 in the backlog is likely what you are looking for. Be sure to add your 👍 if you feel that feature is important.

**Reply by mariXanna** [reply]
Thanks! This is exactly what we need (voted up). 
Does it make sense to submit software to winget if it cannot be unistalled without user interaction ? (until this feature will be ready)

**Reply by denelon** [reply]
@mariXanna we don't have any kind of equivalent for uninstalling a collection of packages. The main reason for the requirement on silent install is to support unattended installs for a collection of packages. I think interactive uninstall is a current necessity. If we were to require silent uninstall, we would likely have to remove several packages, and I think at this point it would be a terrible user experience to have a bunch of packages suddenly become unavailable.
